Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -681,15 +637,12 @@ namespace OpenInfraPlatform {
 						double b = i_shape->OverallWidth * length_factor;
 						double tw = i_shape->WebThickness * length_factor;
 						double tf = i_shape->FlangeThickness * length_factor;
-						double r = 0;
-						if(i_shape->FilletRadius) {
-							r = i_shape->FilletRadius * length_factor;
-						}
+						double r = i_shape->FilletRadius.value_or(0.0) * length_factor;
 
 						outer_loop.push_back(carve::geom::VECTOR(b * 0.5, -h * 0.5));
 						outer_loop.push_back(carve::geom::VECTOR(b * 0.5, (-h * 0.5 + tf)));
 
-						if(r != 0) {
+						if(r != 0.0) {

```

Comment:
Don't compare doubles using normal == or != operator. See https://stackoverflow.com/questions/12278523/comparing-double-values-in-c/35252979

No need to change that here now, just for future implementations!
We should also maybe provide that functionality via BlueFramework or some other utility.

Reply:
See #148 .